Sources/ApolloCodegenLib/FileGenerators/OperationManifestFileGenerator.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+/// Representation of an operation that supports Automatic Persisted Queries
+struct OperationManifestItem {
+  let name: String
+  let identifier: String
+  let source: String
+  let type: CompilationResult.OperationType
+
+  init(operation: IR.Operation) {
+    self.name = operation.definition.name
+    self.identifier = operation.operationIdentifier
+    self.type = operation.definition.operationType
+
+    var source = operation.definition.source
+    for fragment in operation.referencedFragments {
+      source += "\n\(fragment.definition.source)"
```

Comment:
This is precisely the same way that the document is assembled for actual execution? Any chance it can share the code with it?

Reply:
(And is the ordering of the fragments well-defined?)

Reply:
I did think about that, but the codegen library and the actual client library don't currently have any shared code. This is done exactly the same in both cases, and there are unit tests for that.

The ordering of fragments is stable ("well-defined" is questionable, as we don't really have this documented anywhere). The order that fragments are first encountered as the operation is compiled is the order they will be listed. So changing the operation itself and re-ordering selections would potentially change this, but given the exact same inputs, it will always be the same. The `referencedFragments` are stored as an `OrderedSet`, so they are guaranteed to retain their ordering (unlike a `Set` which is a random access collection).

Also, this same `referencedFragments` ordered set is used to generate the operation definition metadata that is used by the client to build the HTTP request. So if the order of fragments changed in code gen, then the order of the fragments in the HTTP request would naturally change in the same way.

Reply:
@AnthonyMDev That's fair, though I might suggest a pair of comments in the implementations noting that there is a second implementation of the same document-assembly code that should be kept in sync.

For example if you decided to add a trailing newline or preceding comment or something to the entire operation, it would be ideal to be consistent.

(Also, does this need to be adjusted for the single- vs multi- line thing, or is that at another level?)

Reply:
That's a good callout. We can do that. I think the actual appropriate way to do this is to write a unit test or two that confirm the behavior is the same! :)

(The mult-line thing is seperate from this, thanks for making me think about it and confirm though!)

Reply:
(So single-line is really "each top-level definition is one line"?)